### PR TITLE
Add HUDDrawKeyMenu hook

### PR DESCRIPTION
**Pull request by Partixel**
Allows adding custom buttons to key menu

### CODE REVIEW

**Comment by Partixel** [comment]
There you go @HDG-MrDeeJay

**Comment by HDG-MrDeeJay** [comment]
No yelling from me.

File: entities/weapons/keys/cl_menu.lua [hunk 1]
```diff
@@ -1,3 +1,22 @@
...
+	parameters = {
+		{
+			name = "ent",
+			description = "The door entity.",
+			type = "Entity"
+		},
+		{
+			name = "Frame",
+			description = "The key menu frame.",
+			type = "table" -- I ASSUME ITS A TABLE ?
```
Comment (Bo98) [inline entities/weapons/keys/cl_menu.lua]: Panel.

**Comment by Partixel** [comment]
I caught the error before FPtje saw it this time, :P

**Comment by Partixel** [comment]
Okay Bo.

**Comment by FPtje** [comment]
They keys menu is not a HUD thing, nor a drawing thing. It's a gui thing.

**Comment by FPtje** [comment]
I'll think of a better name when I merge it, though.

**Comment by Partixel** [comment]
Okay :). How about, KeyMenuOpened?

**Comment by Partixel** [comment]
Any reason this hasn't been merged yet?

**Comment by FPtje** [comment]
haven't had time yet

**Comment by Partixel** [comment]
Fair enough, I thought you may have been waiting for me to do something :P.

**Comment by Partixel** [comment]
Glad to see I did it mostly right. Is there any documentation on the hookStub system btw?

**Comment by FPtje** [comment]
No, it's pretty straight forward.

**Comment by Bo98** [comment]
Documentation for documentation.

**Comment by Partixel** [comment]
From what I can tell, does it simply let coders know what to expect argument / return wise or does it actually serve a function?

**Pull request merged**